crates/lib/src/api.rs

```diff
@@ -311,3 +316,53 @@ pub async fn retrieve_results(
         .map(ExecutionResults::from)
         .ok_or_else(|| GrpcClientError::ResponseEmpty("Controller Job Execution Results".into()))
 }
+
+/// API Errors encountered when trying to list available quantum processors.
+#[derive(Debug, thiserror::Error)]
+pub enum ListQuantumProcessorNamesError {
+    /// Failed an http call
+    #[error("Failed an http call: {0}")]
```

Comment:
This should be more descriptive, something like "List processors API returned an error". Not sure if there should be a hyphen in "List(-)processors" or not.